src/resource-graph/azext_resourcegraph/_validators.py

```diff
@@ -7,8 +7,11 @@
 
 
 def validate_query_args(namespace):
-    if not 1 <= namespace.first <= 5000:
-        raise CLIError("Value of --first has to be between 1 and 5000.")
+    if namespace.first and not 1 <= namespace.first <= 1000:
+        raise CLIError("Value of --first has to be between 1 and 1000.")
 
-    if not namespace.skip >= 0:
+    if namespace.skip and not namespace.skip >= 0:
         raise CLIError("Value of --skip cannot be negative.")
+
+    if namespace.subscriptions is not None and namespace.management_groups is not None:
+        raise CLIError("Both --subscriptions and --management-groups cannot be passed together at the same time.")
```

Comment:
Please follow the [guideline](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md) to deprecate CLIError.